src/uct/tcp/tcp_ep.c

```diff
@@ -257,6 +257,14 @@ static UCS_CLASS_CLEANUP_FUNC(uct_tcp_ep_t)
         ucs_free(put_comp);
     }
 
+    if ((self->flags & UCT_TCP_EP_FLAG_FAILED) &&
+        (self->failed_cb_id != UCS_CALLBACKQ_ID_NULL)) {
+        /* a failed EP callback is still scheduled on UCT worker, unregister
+         * it to prevent a callback is being invoked for the destroyed EP */
+        uct_worker_progress_unregister_safe(&iface->super.worker->super,
+                                            &self->failed_cb_id);
```

Comment:
minor: ```cb_id != UCS_CALLBACKQ_ID_NULL``` is already checked in ```uct_worker_progress_unregister_safe``` and you may skip this evaluation here

Reply:
fixed